### DISCUSSION TITLE
Answers for the challenges

### DISCUSSION THREAD

**Discussion by lightydo** [CleanCut/ultimate_rust_crash_course#25] [General]
Where can I find answers to the challenges?

**Comment by CleanCut** [comment]
There aren't any canonical answers for the challenges! (That's why they're _a challenge_).

Though students have asked about many of the challenges [on Udemy in the Q&A section](https://www.udemy.com/course/ultimate-rust-crash-course/learn/lecture/18950234#questions) or shared their own solution(s).

If you have any questions about specific challenges, I'd be happy to answer them here as well!

**Comment by lightydo** [comment]
Thanks, Nathan; I just wanted to check if there are more idiomatic ways to answer the challenges.

**Reply by CleanCut** [reply]
[Ultimate Rust 2: Intermediate Concepts](https://www.udemy.com/course/ultimate-rust-2/?referralCode=8ED694EBE5637F954414) focuses heavily on the idiomatic way to do things. 😉 

And, of course, I'm happy to answer any specific questions you have.

**Comment by Abvit-eve** [comment]
Just found this and plan on signing up for your course.  Great content! 

My question is related to the first exercise. For the constants and variables, you set them to i32. Would it be better to set them to u8? I come from the microcontroller world where it's very resource constrained. I am a product guy trying to teach myself more code.

**Reply by CleanCut** [reply]
Good question.

If you are unable to store two 32-bit integers, you are definitely dealing with a specialized embedded system of some sort! 😆 

_Generally_ (read consumer & server hardware), 32-bit integers are the most performance-optimized integer, even on 64-bit architectures, which is why `i32` is the default integer type in Rust. The fact that it is the default integer type is why it is used in the first tutorial.

Another reason I use `i32` initially is that I want the student to be able to be able to practice type inference (not explicitly specifying the type), but other than taking the default integer type, I haven't taught the student anything that the compiler can use to infer the type, yet.

You can, of course, try any other type like `u8` if you like. You will just need to specify the type explicitly in all declarations.
